test/NeoSharp.Core.Test/ExtensionMethods/StringExtensionMethods.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections.Generic;
+using Microsoft.Extensions.Configuration;
+using NeoSharp.Core.Network;
+using NeoSharp.Core.Network.Security;
+
+namespace NeoSharp.Core.Test.ExtensionMethods
+{
+    public static class StringExtensionMethods
+    {
+        public static NetworkConfig GetNetworkConfig(this string[] peerEndPoints)
```

Comment:
Theoretically, I can get network configuration from any array of strings 🙄 Being serious I suggest to revert this change back to a local method. If it is absolutely unacceptable maybe do it like this:
```
AutoMockContainer.RegisterNetworkConfig(peerEndPoint);
```

Reply:
Having static methods in a class is a code smell ..

* https://searchwindevelopment.techtarget.com/tip/Static-methods-are-a-code-smell
* https://www.c-sharpcorner.com/article/static-variable-a-code-smell/
* http://www.markschabacker.com/blog/2015/03/28/code-smell-statics/

From one of the links I will extract this statement;
The performance improvement for instance to static refactoring is almost never enough to justify this change. Specifically, the following questions are useful when considering this type of refactoring:

Will I ever have the need to extract an interface for this class’s behavior?
Will I ever need to extend this functionality? Perhaps in a subclass?
Will I ever need to stub or mock this behavior?

Reply:
Let me answer you by bringing up the next points:
1. Extension methods that are business context specific on base types are the code smell.
2. I suggested introducing RegisterNetworkConfig extension method on context-specific AutoMockContainer class.
3. In articles, you mentioned, authors referred to static methods/variables that are a part of the public contract of the class. There is nothing wrong to have __private__ static method/variable, especially, in unit tests.